File: javascript-exercises-master/pig_latin/pigLatin.js
function translate(str) {
	return str.
        replace(/\b([aeiou]+)[^aeiou ]+[A-Za-z]*/gi, "$&ay").
        replace(/\b([^aeiou ]?[q][u]|[^aeoui ]+)([A-Za-z][^. ]*)/gi, "$2$1ay");
}

module.exports = {
	translate
};

File: web-dev-101/learn-ruby/04_pig_latin/pig_latin.rb
def translate(string)
  string.gsub(/(\b([aeiou]+)[^aeiou ]+[A-Za-z]*)/i, '\1ay')
        .gsub(/\b([^aeiou ]?[q][u]|[^aeoui ]+)([A-Za-z][^. ]*)/i, '\2\1ay')
end
